code/modules/mob/living/silicon/robot/drone/drone_items.dm

```diff
@@ -141,22 +193,99 @@
+	//Temporary storage to assess amount gained per use
+	var/list/store_buffer = list(
+		"metal" = 0,
+		"glass" = 0,
+		"wood" = 0,
+		"plastic" = 0,
+		"plasteel" = 0,
+		"biomass" = 0
+		)
+	//Metal, glass, wood, plastic, plasteel, biomass
 	var/list/stored_comms = list(
 		"metal" = 0,
 		"glass" = 0,
 		"wood" = 0,
-		"plastic" = 0
+		"plastic" = 0,
+		"plasteel" = 0,
+		"biomass" = 0
 		)
 
+/obj/item/weapon/matter_decompiler/verb/check_salvage()
+
+	set name = "Check Salvage Levels"
+	set desc = "Review levels of stored salvage."
+	set category = "Robot Commands"
+	var/S = null
+	var/amount = null
+
+	for(S in src.stored_comms)
+		amount = src.stored_comms[S]
+		src.loc << "\blue The decompiler has [amount] units of [S] stored."
+
+
+/obj/item/weapon/matter_decompiler/verb/recompile_matter()
+
+	set name = "Recompile Matter"
+	set desc = "Recompiles stored matter into sheets."
+	set category = "Robot Commands"
+	var/stored = null
+	var/stackamount = null
+
+	var res_type = input("Select a resource to be recompiled:", "Robot", null, null) in src.stored_comms
+	if(res_type)
+		stored = src.stored_comms[res_type]
+		stackamount = round(input("You currently have adequate resources to produce [stored] sheets. How many sheets of [res_type] do you want to produce?") as num)
+	if(stackamount <= 0 || stored <= 0)
+		return
+	if(stackamount > stored) //If the stated amount is greater than biomass actually in the stack
+		stackamount = round(stored)
+
+	//Determine material
+	var/material
+	switch(res_type)
+		if("metal")
+			material = /obj/item/stack/sheet/metal
+			recompile_stacks(stackamount, material, stored, 50, res_type)
+		if("glass")
+			material = /obj/item/stack/sheet/glass
+			recompile_stacks(stackamount, material, stored, 50, res_type)
+		if("plastic")
+			material = /obj/item/stack/sheet/mineral/plastic
+			recompile_stacks(stackamount, material, stored, 50, res_type)
+		if("wood")
+			material = /obj/item/stack/sheet/wood
+			recompile_stacks(stackamount, material, stored, 50, res_type)
+		if("plasteel")
+			material = /obj/item/stack/sheet/plasteel
+			recompile_stacks(stackamount, material, stored, 50, res_type)
+		if("biomass")
+			material = /obj/item/stack/sheet/biomass
+			recompile_stacks(stackamount, material, stored, 300, res_type)
+
+
+/obj/item/weapon/matter_decompiler/proc/recompile_stacks(var/stackamount, var/obj/item/stack/sheet/material, var/stored, var/stack_max, var/res_type)
+	src.loc.visible_message("\red [src.loc] recompiles [stackamount] sheets of [res_type]","\blue You recompile [stackamount] sheets of [res_type]",)
+	while(stackamount > 0)
+		var/obj/item/stack/S = PoolOrNew(material,get_turf(src))
+		S.amount = min(stack_max,stackamount)
+		stored -= min(stack_max,stackamount)
+		stackamount -= min(stack_max,stackamount)
+
+	playsound(src.loc, 'sound/weapons/wave.ogg', 50, 1)
+	src.stored_comms[res_type] = stored
+	src.loc << "\blue You have [src.stored_comms[res_type]] units of [res_type] remaining."
```

Comment:
Again with the src. calls, these are technically acceptable, but really not necessary and just look bad.